### PR TITLE
add the model

### PR DESCRIPTION
**Pull request by NickEmma**
# Hi there 👋🏻

**Thanks for taking the time to review the project. Here are some things important to note about it:**

### Changes implemented in the "Model Processing" branch

- ✅ Follow ERD to create associations between models.
- ✅ Create examples for the database and test CRUD methods.
- ✅ Create additional methods for each model.

# Final notes 📔
Is expected that this project meets the requirements. Please let me know if anything is left to be done or can be improved. Again, thanks for the review.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    user.increment!(:post_counter)
+  end
```
Comment (gemmen29) [inline app/models/post.rb:8]: - [ ] [OPTIONAL] Kindly use `after_save` to call `update_posts_counter` and male the `update_posts_counter` method private. The `after_save` will call this method each time a new post is saved in the database.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_likes_counter
+    post.increment!(:likes_count)
+  end
```
Comment (gemmen29) [inline app/models/like.rb:7]: - [ ] [OPTIONAL] Please do the same here.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_comments_counter
+    post.increment!(:comments_count)
+  end
```
Comment (gemmen29) [inline app/models/comment.rb:7]: - [ ] [OPTIONAL] Please do the same here.

**Review by gemmen29 — APPROVED** [review]
Hi @NickEmma, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions:
- [x] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 3 above)
Comment (NickEmma) [reply app/models/comment.rb:7]: thanks

File: app/models/like.rb (hunk 2 above)
Comment (NickEmma) [reply app/models/like.rb:7]: thank you

Comment (NickEmma) [reply app/models/like.rb:7]: thank you

File: app/models/post.rb (hunk 1 above)
Comment (NickEmma) [reply app/models/post.rb:8]: noted

**Pull request merged**